theano/tests/unittest_tools.py

```diff
@@ -445,3 +445,10 @@ def attempt_multiple_times(*args, **kwargs):
                         current_seed = str(int(current_seed) + 1)
 
         return attempt_multiple_times
+
+
+def expectedFailure_fast():
+    """A Decorator to handle the test cases that are failing when
+    THEANO_FALGS =cycle_detection='fast'.
+    """
+    return unittest.expectedFailure if theano.config.cycle_detection == 'fast' else lambda x: x
```

Comment:
It is a bit strange to have to call this function each time, so that it returns the actual decorator.
IMO it would be a bit more simpler to define it as:
```python
def expectedFailure_fast(f):
    if theano.config.cycle_detection == 'fast':
        return unittest.expectedFailure(f)
    else:
        return f
```
Then just use `@expectedFailure_fast` instead of `@expectedFailure_fast()`